planner/core/task.go

```diff
@@ -1196,10 +1253,99 @@ func (p *PhysicalTopN) pushTopNDownToDynamicPartition(copTsk *copTask) (task, bo
 	return attachPlan2Task(p, rootTask), true
 }
 
+func (p *PhysicalTopN) checkOrderPropForSubIndexScan(idxCols []*expression.Column, idxColLens []int, constColsByCond []bool, colsProp *property.PhysicalProperty) bool {
```

Comment:
Please add some comments for these helper functions ... `checkOrderPropForSubIndexScan`, `checkSubScans`, `addPartialLimitForSubScans`.